cli/configuration/v1/flat_map.go

```diff
@@ -0,0 +1,39 @@
+package v1
+
+import (
+	"slices"
+)
+
+// FlatMap merges the provided configs into a single config.
+// The configurations are merged in the order they are provided.
+// Nested configurations are flattened into a single configuration.
+// Configuration types are decoded the least effort, and if they are not yet decoded,
+// they will only be loaded in if they are of type ConfigType.
+// All other types will be left as is and taken over.
+func FlatMap(configs ...*Config) (*Config, error) {
+	merged := new(Config)
+	merged.Configurations = make([]Configuration, 0)
+	for _, config := range configs {
+		flattenCandidates := make([]*Config, 0)
+		for _, config := range config.Configurations {
+			var cfg Config
+			if err := scheme.Convert(config.GetType(), &cfg); err != nil {
+				merged.Configurations = append(merged.Configurations, config)
+			} else {
+				flattenCandidates = append(flattenCandidates, &cfg)
+			}
+		}
+
+		cfg, err := FlatMap(flattenCandidates...)
+		if err != nil {
+			return nil, err
```

Comment:
`FlatMap` does not return any error, except this one. Thus, we do not need to return an error.

Reply:
You're right.